core/savings.go

```diff
@@ -5,82 +5,130 @@ import (
 	"time"
 
 	"github.com/benbjohnson/clock"
+	"github.com/evcc-io/evcc/tariff"
 	"github.com/evcc-io/evcc/util"
 )
 
+// publisher gives access to the site's publish function
+type publisher interface {
+	publish(key string, val interface{})
+}
+
 // Site is the main configuration container. A site can host multiple loadpoints.
 type Savings struct {
-	log *util.Logger
-
+	log                    *util.Logger
+	clock                  clock.Clock
+	tariffs                tariff.Tariffs
 	started                time.Time // Boot time
 	updated                time.Time // Time of last charged value update
-	chargedTotal           float64   // Energy charged since startup (kWh)
+	chargedGrid            float64   // Grid energy charged since startup (kWh)
```

Comment:
Naming: 


    gridCharged

Reply:
Das würde ich gerne so rum lassen. Hatte das erst anders herum und finde die Zusammengehörigkeit so besser ersichtlich. Also als erstes "Was wir erfassen" und als zweites "Welche Art/Anteil". Das ist jetzt konsistent für charged und cost bis in die UI so umgesetzt.

Reply:
Ich habs nun überall umgedreht. Also zuerst `grid/selfConsumption/total` dann den Inhalt.